imaging/ml/ml_codelab/scripts/inference/inference.py

```diff
@@ -526,7 +526,7 @@ def _PubsubCallback(self, message):
     Args:
       message: Incoming pubsub message.
     """
-    image_instance_path = message.data
+    image_instance_path = message.data.decode()
```

Comment:
@codezizo In our code we decode only for PY3 in a similar spot. I don't fully understand why, but I'd expect the same to be necessary here (otherwise we should update our code in pubsub_format.py)?

Reply:
.decode() on a string/byte in python changes it to type unicode which can cause problems. vs in python 3 bytes .decode -> string.  So I would make not decode for python 2. Something like this:
```
if six.PY2:
  image_instance_path = message.data
else:
  image_instance_path = message.data.decode()

```

Reply:
To be fair, in all other spots we do decode() in both PY2 and PY3 without any problems... I am not sure why we have an 'if' in this spot - maybe it's a re2 library...

Reply:
> To be fair, in all other spots we do decode() in both PY2 and PY3 without any problems... I am not sure why we have an 'if' in this spot - maybe it's a re2 library...

i guess it should some low-level string manipulations to face any problems. i didn't meet any problems as well

Reply:
Yes perhaps its just a problem with re2. However I just tested the re library and it looks like it works regardless of unicode. Therefore we can keep it as it was.